### ISSUE TITLE
Enhancement request: (maybe for Pro) Allow user-selection of preferred search results order

### ISSUE THREAD

**Issue by jaysmithgit** [RealRaven2000/QuickFolders#157]
I was going to add a comment to Issue 82, however, that was closed some time ago.

Perhaps my enhancement request could be a feature in Pro, in the "Advanced Search Settings" dialog.

I am finding that the order of search results is not very useful to **me**.

I am not suggesting that your current native method should be changed, but instead that additional options be available to users (Pro?).

I would like to see the results presented in the "native" order that the Thunderbird "left side folder pane" presents them.  In my case, that is alphabetical -- within the context of our selected LOCALE setting that controls sorting.  If Thunderbird is presenting them in a certain order, it seems like it "should" be fairly straight-forward to use the same approach they use.  

( I know that there are add-ons that can provide custom sorting/presentation in Thunderbird; I am not sure what to do if a user is also using an add-on that does that.)

I don't know what all the possible options for sorting might be, but all I care about **in the search results** is strict alpha (ASCII sort) order.  Perhaps other users would prefer something else.  Another alternative might be "most recently accessed" order.

Often what I am searching for are peoples' names (we have over 10,000 folders with clients' names).  For a common name, like "anderson" I get a very long list of search results that is too long to read through quickly because it is currently appears in what looks to me like random order (I am sure there is logic to it, but I can't figure out what that logic is). Thus I am forced to look for something more specific, such as "anderson_b" [for bob] or "anderson_r" [for robert].  But that is not always as easy as it sounds because somebody could really have a folder name of "anderson_frankin_robert_bob" (because the person does not use the "franklin" part of his name very much; and he prefers to be addressed as "bob").  

So, if I am able to see an alpha-sorted (full ASCII sort in which the undercore sorts BEFORE any letters) list, I can visually scan that in an instant instead of having to study it closely.

**Comment by RealRaven2000** [comment]
> I would like to see the results presented in the "native" order that the Thunderbird "left side folder pane" presents them. In my case, that is alphabetical -- within the context of our selected LOCALE setting that controls sorting. If Thunderbird is presenting them in a certain order, it seems like it "should" be fairly straight-forward to use the same approach they use.
> 

not really, because it the folder tree sorts by parent folders so you can have

s/Smith Ashley
s/Smith Anderson

that;s clearly not in alphabetical order in the tree from the POV of the folder name, because it is determined by the parent folder first. Since the search results in the menu are "flattened" what should the order be based on, the target folders or the parent folders?

What should it do in this case:

root/customers/s/Smith Ashley
root/billing/invoices/companies/Smith Anderson

It's not trivial to flatten a hierarchy and expect alphanumeric sorting. I think just using the actual target folder name and sorting that would be the most intuitive method. I think this is what I do, for example:

![image](https://user-images.githubusercontent.com/6314674/117205015-f4710e80-ade8-11eb-89a2-4a1365d433c4.png)

**Comment by jaysmithgit** [comment]
My intent was to have the OPTION for a very simple method of sorting...

1) Simply make a list of all the paths that qualify.

2) Sort them alpha-numerically (standard ASCII sort) from the left-most character.

Thunderbird sorts the folder tree (as displayed left-hand-folder-pane) in this type of standard sortation.  That is how I would prefer to be able to see the sorted results.  (But that's just me!)

Whether or not one displays the root of the path is a different subject, but *I* would display the root of the path.

In a properly designed (with the proper folder structure naming), this is completely organized and logical. If a sysadmin uses illogical naming in the folder tree, that is beyond any form of help.
